lib/chai/core/assertions.js

```diff
@@ -988,6 +988,13 @@ module.exports = function (chai, _) {
 
   function assertInstanceOf (constructor, msg) {
     if (msg) flag(this, 'message', msg);
+    var isFunction = constructor instanceof Function;
+
+    if (!isFunction) {
```

Comment:
I think we can do

```js
var type = typeof constructor;
if (typeof !== 'function') {
    throw new Error
}
```

Maybe instead of `var type = constructor === null ? 'null' : typeof constructor;` we can use `type-detect` here?